modules/Microsoft.Batch/batchAccounts/deploy.bicep

```diff
@@ -30,19 +30,15 @@ param storageAccessIdentity string = ''
 @description('Optional. The allocation mode for creating pools in the Batch account. Determines which quota will be used.')
 param poolAllocationMode string = 'BatchService'
 
-@description('Conditional. The key vault to associate with the Batch account. Required if the \'poolAllocationMode\' is set to \'UserSubscription\' and requires the service principal \'Microsoft Azure Batch\' to be granted contributor permissions on this key vault.')
-param keyVaultReferenceResourceId string = ''
-
 @description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
 param privateEndpoints array = []
 
 @description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set and networkProfileAllowedIpRanges are not set.')
 @allowed([
-  ''
-  'Enabled'
   'Disabled'
+  'Enabled'
 ])
-param publicNetworkAccess string = ''
+param publicNetworkAccess string = 'Enabled'
```

Comment:
We have a logic that disables the public network access if private endpoints are set and network rules are not set, unless   differently specified. The logic wouldn't work if we enabled the public access by default. Unless there's a specific reason for this change I'd suggest to revert it.